### PR TITLE
[ENG-2000] Adding multilevel reg capabilities

### PR DESCRIPTION
**Pull request by corbinSanders**
## Purpose

Uploading multiple levels of a registration to Internet Archive

## Changes

Adding function for getting and uploading multiple levels

## QA Notes

  - Does this change require a data migration? If so, what data will we migrate? N/A
  - What is the level of risk? Minimal
    - Any permissions code touched? N/A
    - Is this an additive or subtractive change, other? Additive
  - How can QA verify? (Through UI, API, AdminApp or AdminAdminApp?)
    - If verifying through API, what's the new version? Please include the endpoints in PR notes or Dev docs.
  - What features or workflows might this change impact?
  - How will this impact performance?
-->

## Documentation
N/A

## Ticket

https://openscience.atlassian.net/browse/ENG-2000

### CODE REVIEW

**Comment by Johnetordoff** [comment]
My understanding of this ticket was that is was supposed to go node-to-node down a component hierarchy for a registration with components "pigeoning" each node and marking them as related the metadata attributes listed on [this page](https://archive.org/services/docs/api/simplelists.html). So I'm not sure this is really the whole ticket or just part.

File: osf_pigeon/pigeon.py [hunk 1]
```diff
@@ -171,6 +184,44 @@ def main(
...
+    for child_guid in children_guids:
+        child_data = {}
+        child_url = main(
+            child_guid,
+            datacite_username=datacite_username,
+            datacite_password=datacite_password,
+            datacite_prefix=datacite_prefix,
+            ia_access_key=ia_access_key,
+            ia_secret_key=ia_secret_key,
+            multi_level=False
```
Comment (Johnetordoff) [inline osf_pigeon/pigeon.py:216]: This looks like it's getting only the immediate children, Just want to confirm this is the correct behavior.

**Review by Johnetordoff — APPROVED** [review]
Very good, just a minor question.

File: osf_pigeon/pigeon.py (hunk 1 above)
Comment (corbinSanders) [reply osf_pigeon/pigeon.py:216]: /registrations/?filter[root_node]=<guid> grabs all descendants of the specified guid. Children, Grandchildren, etc. The sample archive I sent you has 3 generations.

**Review by brianjgeiger — APPROVED** [review]
This looks good. I wonder if we might, at some point in the future, add the ability to have a human-readable way to navigate the hierarchy. For example, appending to the description a list of children links and a parent link when applicable.

**Pull request merged**